opencog/atoms/base/Atom.h

```diff
@@ -190,6 +195,16 @@ class Atom
         return classserver().isA(at, t);
     }
 
+    /** Returns the outgoing set.
+     *
+     * @return The outgoing set.
+     */
+    virtual inline const HandleSeq& getOutgoingSet() const
+    {
+        static HandleSeq no_atoms;
+        return no_atoms;
+    }
```

Comment:
So the way I looked at it was that calling getName() on a non-Node Atom isn't a programming error, it is new well-defined behavior. Atoms without names will return the empty string for getName(). Those that have names will return the name. 

If you need to know if something has a name or not and don't, perhaps because you are iterating over a bunch of handles in C++ or Atoms in Python, you can use an instance check on the class.